recipes/sml/all/conanfile.py

```diff
@@ -13,37 +13,43 @@ class SMLConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     settings = "compiler"
     no_copy_source = True
+    exports_sources = ['patches/*']
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
-    @property 
-    def _minimum_compilers_version(self): 
+    @property
+    def _minimum_compilers_version(self):
         return {
-            "Visual Studio": "15", 
-            "gcc": "5", 
-            "clang": "5", 
-            "apple-clang": "5.1", 
-        } 
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
 
     def configure(self):
-        if self.settings.compiler.cppstd: 
+        if self.settings.compiler.cppstd:
             check_min_cppstd(self, "14")
-        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False) 
-        if not minimum_version: 
-            self.output.warn("SML requires C++14. Your compiler is unknown. Assuming it supports C++14.") 
-        elif tools.Version(self.settings.compiler.version) < minimum_version: 
-            raise ConanInvalidConfiguration("SML requires C++14, which your compiler does not support.") 
+        minimum_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn(
+                "SML requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                "SML requires C++14, which your compiler does not support.")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
+        tools.patch(patch_file="patches/clang12.patch")
         extracted_dir = "sml-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
```

Comment:
Completely unrelated but this is the new convention

```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```